integration-test/mqtt/mqtt_system_test.c

```diff
@@ -1119,17 +1116,18 @@ void test_MQTT_Restore_Session_Resend_PubRel( void )
 
     /* Disconnect on receiving PUBREC so that we are not able to complete the QoS 2 PUBLISH in the current connection. */
     TEST_ASSERT_FALSE( receivedPubComp );
-    disconnectOnPacketType = MQTT_PACKET_TYPE_PUBREC;
+    packetTypeForDisconnection = MQTT_PACKET_TYPE_PUBREC;
     TEST_ASSERT_EQUAL( MQTTSendFailed,
                        MQTT_ProcessLoop( &context, 2 * MQTT_PROCESS_LOOP_TIMEOUT_MS ) );
     TEST_ASSERT_FALSE( receivedPubComp );
 
-    /* Verify that the connection with the broker has been disconnected. */
-    TEST_ASSERT_EQUAL( MQTTNotConnected, context.connectStatus );
```

Comment:
Why was this removed?

Reply:
As `MQTT_Disconnect` call has been removed, the `MQTTContext.connectedStatus` would not be updated